### ISSUE TITLE
kdtree asserting when removing station

### ISSUE THREAD

**Issue by SamuXarick** [OpenTTD/OpenTTD#7461]
Sorry, this crash triggered when I was testing a custom build… will try to see if I can make it crash in master.

## Version of OpenTTD
2019-04-02-variable-script_max_op_codes-me1aa0e3e71
## Expected result
No crash
## Actual result
crash
## Steps to reproduce

```
 	KernelBase.dll!00007ffc3642a388()	Unknown
 	openttd.exe!raise(int signum) Line 516	C++
 	openttd.exe!abort() Line 71	C++
 	openttd.exe!error(const char * s, ...) Line 126	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 213	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::RemoveRecursive(const ViewportSignKdtreeItem & element, unsigned __int64 node_idx, int level) Line 216	C++
 	openttd.exe!Kdtree<ViewportSignKdtreeItem,int (__cdecl*)(ViewportSignKdtreeItem const &,int),int,int>::Remove(const ViewportSignKdtreeItem & element) Line 414	C++
>	openttd.exe!Station::~Station() Line 167	C++
 	[External Code]	
 	[Inline Frame] openttd.exe!StationHandleBigTick(BaseStation *) Line 3271	C++
 	openttd.exe!OnTick_Station() Line 3694	C++
 	openttd.exe!CallLandscapeTick() Line 1397	C++
 	openttd.exe!StateGameLoop() Line 1375	C++
 	openttd.exe!GameLoop() Line 1463	C++
 	openttd.exe!VideoDriver_Win32::MainLoop() Line 1275	C++
 	openttd.exe!openttd_main(int argc, char * * argv) Line 849	C++
 	openttd.exe!WinMain(HINSTANCE__ * hInstance, HINSTANCE__ * hPrevInstance, char * lpCmdLine, int nCmdShow) Line 443	C++
 	[External Code]	
```

assert(next != INVALID_NODE); // node must exist somewhere and must be found before a leaf is reached
line 213 of kdtree.hpp

**Comment by nielsmh** [comment]
Need a series of steps to trigger the error, or a savegame that will trigger it.

**Comment by SamuXarick** [comment]
Couldn't manage to make it crash in master yet. It was in a all Random AIs + Company Value GS game. Map size wasn't that big, 256x256 if I recall. It crashed rather early too, about a few months into the game.

I'm still trying to make it crash in master.

**Comment by SamuXarick** [comment]
[OpenTTD.zip](https://github.com/OpenTTD/OpenTTD/files/3041289/OpenTTD.zip) - crash files
Crashed again, but not on master. I'm disappointed that I can't be of better help.
EDIT: The exact build was this: https://github.com/SamuXarick/OpenTTD/commit/a702fbc9aa57c5349bad44681818d0a3dded88e8

EDIT2: on the savegame, the station is named Funham Central, it's a demolished station at tile 106034, from white company, RoadAI.

**Comment by nielsmh** [comment]
Likely solved by #7849
